### ISSUE TITLE
differenceInSeconds() returns -0 when comparing the same date with millisecond differences

### ISSUE THREAD

**Issue by ashutoshrishi** [date-fns/date-fns#2555]
```
let d = new Date("2021-07-22T06:01:28.973Z")
let e = new Date("2021-07-22T06:01:28.976Z")

differenceInSeconds(d, e)
// -0

differenceInSeconds(d, d)
// 0
```

I have tested this with the latest version `2.22.1`. There is an older issue and test to prevent the negative 0, so maybe this is a regression?

https://github.com/date-fns/date-fns/blob/master/src/differenceInSeconds/index.ts#L36 

For the above 2 dates, the `differenceInMilliseconds` is `-0.003` and `Math.ceil(-0.003)` is `-0`.

**Comment by pabrodez** [comment]
I wouldn't mind date-fns returning -0 or 0 in the `difference*` functions, but `Object.is(0, -0) === false` : (

**Comment by utluiz** [comment]
I was hit with this issue. Maybe another regression or a different scenario.

```
test('check', () => {
  const dateLeft = new Date(1694143833566)
  const dateRight = new Date(1694143833567)
  const result = differenceInSeconds(dateLeft, dateRight)
  expect(result).toBe(0)
})
```

Results in:

```
Error: expect(received).toBe(expected) // Object.is equality

Expected: 0
Received: -0
```

The issue happens when the `dateLeft` is lower than `dateRight` by an order of magnitude lower than the different unit, so while the result is zero, it still yields a negative result.

**Comment by kossnocorp** [comment]
A PR is welcome

**Comment by kossnocorp** [comment]
Fixed and shipped with v3.3.0: https://github.com/date-fns/date-fns/releases/tag/v3.3.0

**Comment by utluiz** [comment]
Thank you @kossnocorp

FWIW I missed your last update otherwise I'd be glad to git it a shot.

The reason for the `-0` was because the diff computation results in a negative fraction that rounds to zero. Then one of those JS quirks kicks in and the minus sign is preserved.

For instance, you can try something like this in any console:
```javascript
console.log(Math.round(0.001 - 0.002))
-0
```

So effectively, any routine that rounds or ceils fractions can have this issue. 

As for a solution, the workaround I applied to my application as adding a zero:
```javascript
console.log(Math.round(0.001 - 0.002) + 0)
0
```

Basically adding `+0` after calls to `Math.round` and `Math.ceil` would also do it I think.

**Comment by martinhipp** [comment]
Thanks @kossnocorp, any chance of the negative number fixes in v3.3.0 could be backported to v2? Unfortunately I am unable to upgrade to v3, all good if it is too much of an ask, I have fixed it manually, but assume it probably affects a few others as well.
